Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -4297,6 +4297,12 @@ class BBQJIT {
 
         ScratchScope<1, 0> scratches(*this, lhsLocation, rhsLocation, resultLocation);
         if (isSigned && !IsMod && !checkedForNegativeOne) {
```

Comment:
Is this path ever taken?  It looks like `checkedForNegativeOne` is always true by the time it gets here.

Reply:
This path is for the case, both are not constants.

Reply:
Oh right.  I misread the above as an `if-else` block.  It's not.  It's an `if-else if` block.  So, the 2 paths aren't necessarily taken, and `checkedForNegativeOne` can be false.